Question how many <span style='color:orange'>Columns</span> are there in an Excel Sheet?

In [2]:
import numpy as np

In [15]:
alphabets = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm',
             'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z']
abc = np.array(alphabets)

In [4]:
# 1st loop.
all_possible_combinations = abc
for a in abc:
    for b in abc:
        combination = a+b
        all_possible_combinations = np.append(
            all_possible_combinations, combination)


In [5]:
# 2nd loop.
for a in abc:
    for b in abc:
        for c in abc:
            combination = a+b+c
            all_possible_combinations = np.append(
                all_possible_combinations, combination)


In [14]:
print(len(all_possible_combinations))          # Total length.
all_possible_combinations.searchsorted('xfd')  # last column in excel.


18278


16383

<hr>
There are 18278 combination if we combine the combination if two 1+1 (eg: ab) and 1+1+1 three lettes (eg: abc).

Answer: There are <span style='color:green'><b>16384</b></span> coulmns in an Excel sheet.